inference-engine/src/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -169,104 +193,366 @@ class ngraph::pass::mask_propagation::GroupConvolution : public MatcherPass {
     }
 };
 
+class ngraph::pass::mask_propagation::Reshape : public MatcherPass {
+public:
+    Reshape() {
+        auto input = pattern::any_input(pattern::has_static_shape());
+        auto shape = pattern::any_input();
+        auto reshape = pattern::wrap_type<opset6::Reshape>({input, shape});
+
+        ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+            const auto & pattern_map = m.get_pattern_value_map();
+            const auto & m_shape = pattern_map.at(shape);
+            const auto & m_output = pattern_map.at(reshape);
+            const auto & m_input = pattern_map.at(input);
+
+            auto consumers_set = m_output.get_node_shared_ptr()->output(0).get_target_inputs();
+            std::vector<Input<Node>> consumers(consumers_set.begin(), consumers_set.end() );
+
+            // Working only for Reshapes on Group Convolution weights
+            if (!(consumers.size() == 1 && ngraph::is_type<opset6::GroupConvolution>(consumers[0].get_node())
+                    && consumers[0].get_index() == 1)) {
+                std::cout << "HEY";
+                return false;
+            }
```

Comment:
Move to pattern.

Reply:
Deleted as this checks moved to pattern.